### ISSUE TITLE
Exercise 1: HTTP Request Loops (ESLint: no-await-in-loop and ESLint: no-restricted-syntax)

### ISSUE THREAD

**Issue by x2apps** [vkarpov15/mastering-async-await-issues#1]
First, everything works OK except ESLint flags the for loop.

```
for (const post of posts) {
    console.log(`Fetch post ${post.id}`);
    const content = await getWithRetry(`${root}/post?id=${post.id}`, 3);
    if (content.content.includes('async/await hell')) {
      console.log(`Correct answer: ${post.id}`);
      break;
    }
  }
```
ESLint for the above code:
[eslint] iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations. (no-restricted-syntax)
[eslint] Unexpected `await` inside a loop. (no-await-in-loop)

A regular **for** loop eliminates the first ESLint flag, but the second remains. Is there a way to change the code? Searched web but couldn't find anything that would help.

You can add
/* eslint-disable no-await-in-loop,no-restricted-syntax */
to ignore the ESLint.

**Comment by vkarpov15** [comment]
Sounds like an issue with your eslint config, are you using a preset or your own eslint config?

The first `no-restricted-syntax` warning sounds like a false positive. Iterators !== generators, and specifically there's no reason why a transpiler would use regenerator to transpile a `for/of` loop if a generator isn't involved. If you want to use `for (let i = 0; i < posts.length; ++i) {}` to avoid potential transpiler issues, I don't see any problem with that.

The 2nd rule `no-await-in-loop` is missing the point of `await`. Often times you don't want to execute a bunch of requests in parallel to avoid spamming your server/database, and using `await` with a `for` loop is more readable than `await Promise.all(posts.map(async (post) => { /* ... */ }));`. If you're going to do the latter, you might as well go back to using callbacks.

TLDR; I'd change your eslint config to remove the `no-await-in-loop` rule, and if you can't do that I'd just use `/* eslint-disable no-await-in-loop */` because there are [established reasons for why not to use this rule](https://eslint.org/docs/rules/no-await-in-loop#when-not-to-use-it)

**Comment by x2apps** [comment]
Yes, it is the **airbnb** preset that flags the above. Changed to **standard** preset. No flags.

**Comment by vkarpov15** [comment]
Great! I've been debating the merits of the for/of issue with the airbnb devs, should make for some fun reading https://github.com/airbnb/javascript/issues/1271#issuecomment-410087743 .
